src/appleseed/foundation/image/colormap.cpp

```diff
@@ -0,0 +1,187 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2019 Achal Pandey, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "colormap.h"
+
+// appleseed.foundation headers.
+#include "foundation/image/colorspace.h"
+#include "foundation/image/genericimagefilereader.h"
+
+// Standard headers.
+#include <cassert>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <string>
+
+namespace foundation
+{
+
+void ColorMap::set_palette_from_array(const float* values, const size_t entry_count)
+{
+    m_palette.resize(entry_count);
+
+    for (size_t i = 0; i < entry_count; ++i)
+    {
+        m_palette[i] =
+            Color3f(
+                values[i * 3 + 0],
+                values[i * 3 + 1],
+                values[i * 3 + 2]);
+    }
+}
+
+void ColorMap::set_palette_from_image_file(const std::string& filepath)
```

Comment:
This method doesn't belong in this class whose sole purpose is to remap colors given a color palette, not to do I/O of any kind. Move it back to `ColorMapPostProcessingStage` (and remove header files that are no longer necessary).